<a href="https://colab.research.google.com/github/somasekhar1919/Stat-Team-11/blob/main/Copy_of_Z_test_Final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Z TEST FOR ONE SAMPLE
# Group No: **11**
---
---





In [11]:
import pandas as pd
import numpy as np
from statsmodels.stats.weightstats import ztest 

df = pd.read_csv("/content/drive/MyDrive/apy.csv")
df = df.dropna(how='any')

databo = (df['State_Name']== 'Karnataka')   # data filter 
data_karnataka = df.loc[databo, 'Production']   


76865     2602.0
76866    52375.0
76867       61.0
76868    22129.0
76869     7734.0
          ...   
97981       34.0
97982    35029.0
97983        5.0
97984     1499.0
97986    78614.0
Name: Production, Length: 21079, dtype: float64

In [12]:
alpha=0.05 #since our CI is 95%
Production_Mean = np.mean(data_karnataka)
sigma = np.std(data_karnataka)

print("The Mean of Production               =" ,Production_Mean)
print("The Standard Deviation of Production =" ,sigma)


The Mean of Production               = 40961.61163622562
The Standard Deviation of Production = 426983.89900100586


# Production Mean = 40962

#      H0: Production is 40962

#      H1: Production is not equal to 40962

In [14]:
sample_size = 2500
np.random.seed(10)
a = np.random.choice(data_karnataka,sample_size)
a

array([1.44000e+02, 1.65000e+02, 3.02671e+05, ..., 7.20000e+02,
       3.70000e+01, 1.18390e+04])

In [15]:
a_mean = np.mean(a)
a_mean

55003.116264

In [18]:
n = sample_size
Std_1 = sigma/np.sqrt(n)
Std_1


8539.677980020117

In [19]:
data = Std_1*a+Production_Mean
data

array([1.27067524e+06, 1.45000848e+06, 2.58475384e+09, ...,
       6.18952976e+06, 3.56929697e+05, 1.01142209e+08])

In [24]:
ztest_score,p_value = ztest(data,value=a_mean)

print(ztest_score,p_value)

4.445155578913273 8.78282844945009e-06


In [21]:
if(p_value <  alpha):
  print("Reject Null Hypothesis")
else:
  print("Fail to Reject NUll Hypothesis")

Reject Null Hypothesis
